ddsrecorder/test/blackbox/mcap/McapFileCreationTest.cpp

```diff
@@ -0,0 +1,712 @@
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::STOPPED,
+        n_data_1, n_data_2,
+        McapHandlerState::RUNNING,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, (n_data_2));
+
+}
+
+TEST(McapFileCreationTest, transition_paused_stopped)
+{
+    std::string file_name = "output_transition_paused_stopped_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::PAUSED,
+        n_data_1, n_data_2,
+        McapHandlerState::STOPPED,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, 0);
+
+}
+
+TEST(McapFileCreationTest, transition_stopped_paused)
+{
+    std::string file_name = "output_transition_stopped_paused_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::STOPPED,
+        n_data_1, n_data_2,
+        McapHandlerState::PAUSED,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, 0);
+
+}
+
+TEST(McapFileCreationTest, transition_running)
+{
+    std::string file_name = "output_transition_running_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::RUNNING,
+        n_data_1, n_data_2,
+        McapHandlerState::RUNNING,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, (n_data_1 + n_data_2));
+
+}
+
+TEST(McapFileCreationTest, transition_paused)
+{
+    std::string file_name = "output_transition_paused_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::PAUSED,
+        n_data_1, n_data_2,
+        McapHandlerState::PAUSED,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, 0);
+
+}
+
+TEST(McapFileCreationTest, transition_stopped)
+{
+    std::string file_name = "output_transition_stopped_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::STOPPED,
+        n_data_1, n_data_2,
+        McapHandlerState::STOPPED,
+        0);
+
+    unsigned int n_received_msgs = std::get<0>(recording);
+
+    ASSERT_EQ(n_received_msgs, 0);
+
+}
+
+TEST(McapFileCreationTest, transition_paused_event_less_window)
```

Comment:
This test can fail due to two race conditions.
1. If the time between trigger call and measuring current_time is greater than the reception latency of the first sample (it would be stored in the mcap but would be considered outdated according to current_time).
2. Same would happen if the time between measuring the current time within trigger call is greater than the reception latency of the first sample.

The first one can be solved by measuring current_time before calling trigger. The second is harder to tackle, I cannot think of a solution a the moment, but this race condition is very unlikely so I wouldn't bother.